climate-monitoring-server/src/main/java/climatemonitoring/Skeleton.java

```diff
@@ -1,6 +1,346 @@
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case ADD_AREA:
+
+						Area area = (Area) m_in.readObject();
+
+						try {
+
+							m_serverDatabase.addArea(area);
+
+							m_out.writeObject(true);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case ADD_CENTER:
+
+						Center center = (Center) m_in.readObject();
+
+						try {
+
+							m_serverDatabase.addCenter(center);
+
+							m_out.writeObject(true);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case ADD_OPERATOR:
+
+						Operator operator = (Operator) m_in.readObject();
+
+						try {
+
+							m_serverDatabase.addOperator(operator);
+
+							m_out.writeObject(true);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case ADD_PARAMETER:
+
+						Parameter parameter = (Parameter) m_in.readObject();
+
+						try {
+
+							m_serverDatabase.addParameter(parameter);
+
+							m_out.writeObject(true);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case EDIT_OPERATOR:
+
+						String userID = (String) m_in.readObject();
+						Operator newOperator = (Operator) m_in.readObject();
+
+						try {
+
+							m_serverDatabase.editOperator(userID, newOperator);
+
+							m_out.writeObject(true);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case VALIDATE_CREDENTIALS:
+
+						String loginID = (String) m_in.readObject();
+						String loginPassword = (String) m_in.readObject();
+
+						try {
+
+							Operator result = m_serverDatabase.validateCredentials(loginID, loginPassword);
+
+							m_out.writeObject(true);
+							m_out.writeObject(result);
+						}
+
+						catch (DatabaseRequestException e) {
+
+							m_out.writeObject(false);
+							m_out.writeObject(e.getMessage());
+						}
+
+						break;
+
+					case PING:
+
+						long userTime = m_in.readLong();
+
+						Long result = System.currentTimeMillis() - userTime;
+						m_out.writeObject(result);
+
+						break;
+
+					case DISCONNECT:
+
+						m_running = false;
+
+						break;
+
+					default:
+
+						Console.write("Unhandled request type: " + request);
+
+						break;
+				}
+			}
+
+			catch (IOException | ConnectionLostException e) {
+
+				Console.write("Client disconnected unexpectedly");
+				m_running = false;
+			}
+
+			catch (ClassNotFoundException ex) {
+
+				Console.write("Class of a serialized object cannot be found");
+				m_running = false;
+			}
+		}
+
+		close();
+	}
```

Comment:
At line 319, you should add a console message to print that a client has disconnected

Reply:
Fixed